pennylane/math/quantum.py

```diff
@@ -972,6 +975,99 @@ def sqrt_matrix(density_matrix):
+
+    Args:
+        density_matrix (sparse): 2D sparse density matrix of the quantum system.
+
+    Returns:
+       (sparse): Square root of the density matrix. Even for data types like `csr_matrix` or `csc_matrix`, the output matrix is not guaranteed to be sparse as well.
+
+    """
+    if not issparse(density_matrix):
+        raise TypeError(
+            f"sqrt_matrix_sparse only supports sparse matrices, but got {type(density_matrix)}. "
+            "There will be an inevitable performance hit and risk of divergence for non-sparse matrices."
+        )
+    if density_matrix.nnz == 0:
+        return density_matrix
+    return _denman_beavers_iterations(density_matrix, max_iter=100, tol=1e-10)
+
+
+def _inv_newton(M, guess):
+    return 2 * guess - guess @ M @ guess
+
+
+def _denman_beavers_iterations(mat, max_iter=100, tol=1e-13):
+    """Compute matrix square root using the Denman-Beavers iteration.
+
+    Args:
+        mat (sparse): Sparse input matrix
+        max_iter (int): Maximum number of iterations
+        tol (float): Convergence tolerance
+
+    Returns:
+        scipy.sparse.csc_matrix: Square root of the input matrix
+
+    Raises:
+        LinAlgError: If matrix inversion fails
+        ValueError: If NaN values or overflow are encountered during computation
+    """
+    try:
+        mat = csc_matrix(mat)
+        Y = mat
+        Z = sp.sparse.eye(mat.shape[0]).tocsc()
+
+        # Keep track of previous iteration for convergence check
+        Y_prev = None
+        norm_diff = None
+
+        for iter_num in range(max_iter):
+            # Compute next iteration
+            if iter_num < 2:
+                Zinv = spla.inv(Z) if iter_num > 0 else Z
+                Yinv = spla.inv(Y)
+            else:
+                # Take Newton step
+                Zinv = _inv_newton(Z, Zinv)
+                Yinv = _inv_newton(Y, Yinv)
+
+            Y = 0.5 * (Y + Zinv)
+            Z = 0.5 * (Z + Yinv)
+
+            # Check for NaN or infinite values
+            if not (np.all(np.isfinite(Y.data)) and np.all(np.isfinite(Z.data))):
+                raise ValueError(
+                    "Invalid values encountered during computation: nan or inf"
+                    f"Input matrix: {mat.toarray()}"
+                )
+
+            # Check convergence every 10 iterations
+            if iter_num % 10 == 0:
+                if Y_prev is not None:
+                    # Compute Frobenius norm of difference
+                    diff = Y - Y_prev
+                    norm_diff = spla.norm(diff)
+                    if norm_diff < tol:
+                        break
+                Y_prev = Y.copy()
+
+        numerical_error = spla.norm((Y @ Y - mat))
+        if (norm_diff and norm_diff > tol) or numerical_error > tol:
+            raise ValueError(
+                f"Denman Beavers not converged until the end of {max_iter} loops, "
+                f"with last norm error {norm_diff} and compute error {numerical_error}"

```

Comment:
```suggestion
                f"Convergence threshold not reached after {max_iter} iterations, "
                f"with final norm error {norm_diff} and numerical error {numerical_error}"
```
Just some tidying --- I don't think a user needs to see the algorithm, they'll only care whether a result converged or not.